### PR TITLE
add smartPicker for file embeddings

### PR DESCRIPTION
**Pull request by grnd-alt**
resolves: #74 

### smart picker button top right (same icon as in text from material design)
image

### open smart picker selecting files
image

### multiple file embeddings added via smart picker
image

### CODE REVIEW

**Comment by juliushaertl** [comment]
Two comments without looking at the code yet:
- [ ] Are we somehow able to inject this into the main menubar? Otherwise maybe not for this PR, but would we we able to build our own menubar instead?
- [ ] Picking files is just one aspect, the smart picker also allows to pick different links from other providers like tables/deck. We should make use of the second with `getLinkWithPicker` https://github.com/nextcloud/text/blob/main/src/components/Menu/ActionInsertLink.vue#L218-L229
  - I think the file picker also still makes sense so maybe we can have two buttons Insert a file, Insert smart picker embed

**Comment by juliushaertl** [comment]
Looks great otherwise 👍

**Comment by grnd-alt** [comment]
@juliushaertl I've hacked it a little bit so the button is in the tool bar. It is not supported by the lib to add custom elements to that bar. It is though, also required by some other people: https://github.com/excalidraw/excalidraw/issues/7583
so we could think about an upstream contribution, or also do our own toolbar, seems other people have already done it as well. Don't know if it's a good idea to keep my hacked version. Also Excalidraw uses the same Icon for their 'more tools' button as we do for our smart picker button, which we should probably do something about

File: src/App.tsx [hunk 1]
```diff
@@ -72,6 +76,16 @@ export default function App({ fileId, isEmbedded, fileName }: WhiteboardAppProps
 
 	if (excalidrawAPI && !collab) setCollab(new Collab(excalidrawAPI, fileId))
 	if (collab && !collab.portal.socket) collab.startCollab()
+	useEffect(() => {
+		const extraTools = document.getElementsByClassName('App-toolbar__extra-tools-trigger')[0]
+		const smartPick = document.createElement('label')
```
Comment (juliushaertl) [inline src/App.tsx:81]: Can we add a tooltip as well?

File: src/App.tsx [hunk 2]
```diff
@@ -114,7 +128,43 @@ export default function App({ fileId, isEmbedded, fileName }: WhiteboardAppProps
...
+			strokeColor: '#1e1e1e',
+			backgroundColor: 'transparent',
+			fillStyle: 'solid',
+			strokeWidth: 2,
+			strokeStyle: 'solid',
+			roundness: null,
+			roughness: 1,
+			opacity: 100,
+			width: 100,
+			height: 200,
```
Comment (juliushaertl) [inline src/App.tsx:147]:
```suggestion
			width: 400,
			height: 200,
```

File: src/App.tsx [hunk 3]
```diff
@@ -126,6 +176,14 @@ export default function App({ fileId, isEmbedded, fileName }: WhiteboardAppProps
 		)
 	}
 
+	const renderSmartPicker = () => {
+		return (
+			<button className="dropdown-menu-button App-toolbar__extra-tools-trigger" aria-label="Library" aria-keyshortcuts="0" onClick={pickFile} title='Smart Picker'>
```
Comment (juliushaertl) [inline src/App.tsx:181]:
```suggestion
			<button className="dropdown-menu-button App-toolbar__extra-tools-trigger" aria-label="Smart picker" aria-keyshortcuts="0" onClick={pickFile} title='Smart picker'>
```

File: src/App.tsx [hunk 4]
```diff
@@ -126,6 +176,14 @@ export default function App({ fileId, isEmbedded, fileName }: WhiteboardAppProps
...
+				<Icon path={mdiShape} size={1} />
```
Comment (juliushaertl) [inline src/App.tsx:182]: I think I missed how the icon is defined but maybe we can use https://pictogrammers.com/library/mdi/icon/slash-forward-box/ instead which is also what Talk uses these days

Screenshot 2024-08-16 at 20 41 33

**Review by juliushaertl — CHANGES_REQUESTED** [review]
Very nice, works well, just had a few more polishing comments :)

**Review by juliushaertl — APPROVED** [review]

**Pull request merged**